### DISCUSSION TITLE
Diesel only supports tables with primary keys. Table users has no primary key

### DISCUSSION THREAD

**Discussion by t4t5u0** [diesel-rs/diesel#2924] [Q&A]
I have a error when run `diesel migration run`. 

Can you help me?

up.sql

```sql
CREATE TABLE friends (
    id INT NOT NULL AUTO_INCREMENT,
    acctive TEXT NOT NULL,
    pussive TEXT NOT NULl,
    delete_flag BOOLEAN NOT NULL default 0,
    PRIMARY KEY(id)
);
```

Cargo.toml
```toml
diesel = { version = "1.4.8", features = ["mysql"] }
```

**Comment by weiznich** [comment]
Given the provided details I would guess that this is #2786.

**Reply by t4t5u0** [reply]
Thank you for replying.
I was using MySQL 8, but using MySQL 5.7 worked fine. Is MySQL 8 supported?

**Reply by weiznich** [reply]
MySQL 8 is supported as well by diesel. Just to clarify that: This is just a minor bug in the schema inference code, it's nothing fundamental to diesel connecting to a MySQL 8 database.

To add some more details about the issue here as well:

This seems to happen with newer versions of MySQL 8, which means it not something changed with the initial release of MySQL 8, but something that changed on a 8.0.z release. I'm not sure when exactly this changed and what exactly changed. So any help here tracking down the bug is welcome. 
Diesel does execute the following query to receive all primary keys for a given table:

```sql
SELECT information_schema.key_column_usage.column_name
FROM information_schema.key_column_usage
WHERE information_schema.key_column_usage.table_name = 'your_table_name'
  AND information_schema.key_column_usage.table_schema = 'your_database_name'
  AND information_schema.key_column_usage.constraint_name IN(
    SELECT information_schema.table_constraints.constraint_name FROM information_schema.table_constraints
    WHERE information_schema.table_constraints.constraint_type = 'PRIMARY KEY'
  );
```

As I've failed to reproduce the problem locally till now it would be helpful to get the result of that query with your example schema for both the old and the new MySQL version. (Remember to replace the table name and the database name)
Additionally exact information about the used versions could be helpful, so what version did you use exactly, on which platform, how did you install that version, …
